### ISSUE TITLE
[Support] virtual cam starts but doesn't work

### ISSUE THREAD

**Issue by Genius20000** [johnboiles/obs-mac-virtualcam#206]
**Describe the issue**
A installed latest OBS version and Virtualcam plugin, I can see the Start camera on Tools menu, but I cannot see my webcam output nor on Zoom and QuickTime, I also tried on www.webcamtests.com with no luck.

I uninstalled and reinstalled both OBS and Virtualcam, but nothing changed.

**Desktop (please complete the following information):**
 - OS version: macOS Catalina 10.15.6
 - Application (if applicable): Zoom, QuickTime
 - Virtualcam version: latest
 - OBS version: latest

**Additional context**
LOG

15:01:17.978: CPU Name: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
15:01:17.978: CPU Speed: 2200MHz
15:01:17.978: Physical Cores: 6, Logical Cores: 12
15:01:17.978: Physical Memory: 16384MB Total
15:01:17.978: OS Name: Mac OS X (NSMACHOperatingSystem)
15:01:17.978: OS Version: Version 10.15.6 (Build 19G2021)
15:01:17.978: Kernel Version: 19.6.0
15:01:17.981: hotkeys-cocoa: Getting keyboard keys failed
15:01:17.981: hotkeys-cocoa: Getting keyboard keys failed
15:01:17.981: hotkeys-cocoa: Getting keyboard keys failed
15:01:17.981: hotkeys-cocoa: Getting keyboard keys failed
15:01:17.981: hotkeys-cocoa: Using layout 'com.apple.keylayout.Italian-Pro'
15:01:17.982: Portable mode: false
15:01:18.110: OBS 25.0.8 (mac)
15:01:18.110: ---------------------------------
15:01:18.111: ---------------------------------
15:01:18.111: audio settings reset:
15:01:18.111: 	samples per sec: 44100
15:01:18.111: 	speakers:        2
15:01:18.113: ---------------------------------
15:01:18.113: Initializing OpenGL...
15:01:18.146: Loading up OpenGL on adapter ATI Technologies Inc. AMD Radeon Pro 555X OpenGL Engine
15:01:18.146: OpenGL loaded successfully, version 4.1 ATI-3.10.16, shading language 4.10
15:01:18.243: ---------------------------------
15:01:18.243: video settings reset:
15:01:18.243: 	base resolution:   1440x900
15:01:18.243: 	output resolution: 1440x900
15:01:18.243: 	downscale filter:  Bicubic
15:01:18.243: 	fps:               60/1
15:01:18.243: 	format:            NV12
15:01:18.243: 	YUV mode:          601/Partial
15:01:18.244: NV12 texture support not available
15:01:18.244: Audio monitoring device:
15:01:18.244: 	name: Predefinito
15:01:18.244: 	id: default
15:01:18.244: ---------------------------------
15:01:18.247: Failed to load 'en-US' text for module: 'decklink-ouput-ui.so'
15:01:18.256: os_dlopen(libpython3.7m.dylib->libpython3.7m.dylib): dlopen(libpython3.7m.dylib, 257): image not found
15:01:18.256: 
15:01:18.256: [Python] Could not load library: libpython3.7m.dylib
15:01:18.352: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed
15:01:18.353: No blackmagic support
15:01:18.424: [VideoToolbox encoder]: Adding VideoToolbox H264 encoders
15:01:18.500: [obs-browser]: Version 2.8.6
15:01:18.530: VLC found, VLC video source enabled
15:01:18.531: Failed to load 'en-US' text for module: 'obs-mac-virtualcam.so'
15:01:18.531: [obs-mac-virtualcam] version=1.2.0
15:01:18.531: ---------------------------------
15:01:18.531:   Loaded Modules:
15:01:18.531:     obs-mac-virtualcam.so
15:01:18.531:     vlc-video.so
15:01:18.532:     text-freetype2.so
15:01:18.532:     rtmp-services.so
15:01:18.532:     obs-x264.so
15:01:18.532:     obs-vst.so
15:01:18.532:     obs-transitions.so
15:01:18.532:     obs-outputs.so
15:01:18.532:     obs-libfdk.so
15:01:18.532:     obs-filters.so
15:01:18.532:     obs-ffmpeg.so
15:01:18.532:     obs-browser.so
15:01:18.532:     mac-vth264.so
15:01:18.532:     mac-syphon.so
15:01:18.532:     mac-decklink.so
15:01:18.532:     mac-capture.so
15:01:18.532:     mac-avcapture.so
15:01:18.532:     linux-jack.so
15:01:18.532:     image-source.so
15:01:18.532:     frontend-tools.so
15:01:18.532:     decklink-ouput-ui.so
15:01:18.532:     coreaudio-encoder.so
15:01:18.532: ---------------------------------
15:01:18.532: os_dlopen(../obs-plugins/obs-browser->../obs-plugins/obs-browser.so): dlopen(../obs-plugins/obs-browser.so, 257): image not found
15:01:18.532: 
15:01:18.532: ==== Startup complete ===============================================
15:01:18.567: Service '' not found
15:01:18.570: All scene data cleared
15:01:18.570: ------------------------------------------------
15:01:18.629: coreaudio: device 'Iriun Webcam Audio' initialized
15:01:18.629: [Loaded global audio device]: 'Microfono/disp. ausiliario'
15:01:18.651: Switched to scene 'Scena'
15:01:18.651: ------------------------------------------------
15:01:18.651: Loaded scenes:
15:01:18.651: - scene 'Scena':
15:01:18.651:     - source: 'Testo (FreeType 2)' (text_ft2_source_v2)
15:01:18.651: ------------------------------------------------
15:01:18.746: adding 46 milliseconds of audio buffering, total audio buffering is now 46 milliseconds (source: Microfono/disp. ausiliario)
15:01:18.746: 
15:03:37.324: Service '' not found
15:03:37.327: Audio monitoring device:
15:03:37.327: 	name: Predefinito
15:03:37.327: 	id: default
15:03:37.327: Created profile 'Andrea' (clean, Andrea)
15:03:37.327: ------------------------------------------------
15:03:48.537: Starting stream due to OutputTimer
15:03:48.556: ---------------------------------
15:03:48.556: [x264 encoder: 'simple_h264_stream'] preset: veryfast
15:03:48.557: [x264 encoder: 'simple_h264_stream'] settings:
15:03:48.557: 	rate_control: CBR
15:03:48.557: 	bitrate:      2500
15:03:48.557: 	buffer size:  2500
15:03:48.557: 	crf:          0
15:03:48.557: 	fps_num:      60
15:03:48.557: 	fps_den:      1
15:03:48.557: 	width:        1440
15:03:48.557: 	height:       900
15:03:48.557: 	keyint:       250
15:03:48.557: 
15:03:48.571: [CoreAudio AAC: 'simple_aac']: settings:
15:03:48.571: 	mode:          AAC
15:03:48.571: 	bitrate:       160
15:03:48.571: 	sample rate:   44100
15:03:48.571: 	cbr:           on
15:03:48.571: 	output buffer: 1536
15:03:48.571: [rtmp stream: 'simple_stream'] URL is empty
15:03:48.571: [rtmp stream: 'simple_stream'] Connection to (null) failed: -1
15:03:48.574: ==== Streaming Stop ================================================

**Comment by gxalpha** [comment]
Hi, please check two things:
1) In the folder `/Library/CoreMediaIO/Plug-Ins/DAL` (go there via Finder -> Go to folder -> paste this): Is there a `obs-mac-virtualcam.plugin` file?
2) If yes, please open Console.app, search for `obs-mac-virtualcam` and check for logs appearing once you try to start up the camera (in QuickTime for example)

**Comment by Genius20000** [comment]
> Hi, please check two things:
> 
> 1. In the folder `/Library/CoreMediaIO/Plug-Ins/DAL` (go there via Finder -> Go to folder -> paste this): Is there a `obs-mac-virtualcam.plugin` file?
> 2. If yes, please open Console.app, search for `obs-mac-virtualcam` and check for logs appearing once you try to start up the camera (in QuickTime for example)

Thanks for your reply,
last week I broke my mac :( it will be back in a few days, I'll follow your advice ASAP.

Thanks

**Comment by jimmydx** [comment]
Same issue here. Just downloaded the plugin, shows up in OBS but nothing happens anywhere else.

**Comment by Genius20000** [comment]
> Hi, please check two things:
> 
> 1. In the folder `/Library/CoreMediaIO/Plug-Ins/DAL` (go there via Finder -> Go to folder -> paste this): Is there a `obs-mac-virtualcam.plugin` file?
> 2. If yes, please open Console.app, search for `obs-mac-virtualcam` and check for logs appearing once you try to start up the camera (in QuickTime for example)

Done, I've a lot of rows with this:
obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIODevicePropertyLocation

Thanks

**Comment by DigitalDuquette** [comment]
I've the same issue over here on the latest version of macOS with a fresh install following instructions.  

> default	11:33:10.037997-0400	obs	obs-mac-virtualcam(DAL): PlugInMain version=1.2.0
> default	11:33:10.038060-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_QueryInterface
> default	11:33:10.038086-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_Release sRefCount now = 0
> default	11:33:10.038114-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_InitializeWithObjectID self=0x1e129768
> default	11:33:10.038208-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1684629094
> default	11:33:10.038239-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.038265-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1869180523
> default	11:33:10.038290-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.038316-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1885762592
> default	11:33:10.038338-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.652489-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1818850926
> default	11:33:10.652559-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.652607-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1869180523
> default	11:33:10.652648-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.652689-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1937009955
> default	11:33:10.652730-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.652781-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1886156132
> default	11:33:10.652821-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.652863-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Device(36) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1935833461
> default	11:33:10.652900-0400	obs	obs-mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.652937-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.652973-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653054-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653152-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653235-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653327-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653403-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653500-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653614-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653728-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653815-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.653914-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654037-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654083-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654191-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654251-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654330-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654371-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654478-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654525-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654592-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654652-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654728-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654762-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654835-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654895-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654923-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.654994-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655067-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655137-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655184-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655226-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655323-0400	obs	obs-mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData Stream(37) kCMIOObjectPropertyListenerAdded self=0x1e129768 data(int)=1885762592
> default	11:33:10.655404-0400	obs	obs-mac-virtualcam(DAL): Stream unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
> default	11:33:10.655529-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655603-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655643-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655814-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655894-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.655937-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.656002-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIODevicePropertySuspendedByUser
> default	11:33:10.656029-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
> default	11:33:10.656053-0400	obs	obs-mac-virtualcam(DAL): Device unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects
>

**Comment by KallistoX** [comment]
Same issue here, However I can see the OBS Cam in Quicktime but I get a "Access Denied" trying to display it. But I can't see the Cam in WebEx or Zoom. Im on macOS Catalina 10.15.6.
I could provide you with additional informations if necessary.
Thanks

**Comment by gxalpha** [comment]
Hi @jimmydx ,
I also need [these](https://github.com/johnboiles/obs-mac-virtualcam/issues/206#issuecomment-686716631) details for helping you.

@DigitalDuquette and @KallistoX this looks like a codesigning issue with WebEx (although Zoom should work out of the box, make sure you have the latest Zoom version), look [here](https://github.com/johnboiles/obs-mac-virtualcam/wiki/Compatibility) for more details and info on how to solve it.

@Genius20000 are you able to select the camera in QuickTime (New movie recording)?
If no, are there other log outputs (not the ones with `Stream unhandled hasPropertyWithAddress for kCMIOObjectPropertyOwnedObjects`)?

**Comment by john-guerra** [comment]
I was having the same problem (virtual camera menu showing up and working on Quicktime but not zoom), and fixed it by updating Zoom to the latest version.

**Comment by rachelnicole** [comment]
I don't know if this should be a new thread or not, but it works for me on quicktime and zoom now.  Discord doesn't work though.

**Comment by zeusstl** [comment]
Zoom did not allow virtual cameras for a few months bc of some apple restrictions. But if you get the latest mac version of zoom now, you should see the virtual camera.

"Support for all virtual cameras
Zoom will now support all virtual cameras. We no longer require virtual cameras to be approved before they can be used with Zoom."
https://support.zoom.us/hc/en-us/articles/201361963-New-updates-for-macOS

**Comment by nabiliqbal** [comment]
I am finding that Zoom can see the virtual camera but nothing from OBS comes through (it is just a gray screen.) Works however on Google Meet. Any ideas? (I am using latest version of Zoom, 5.3.0).

**Comment by zeusstl** [comment]
The OBS video still comes through for me on version 5.3.0. (on Mojave) (It is extremely laggy, though, which I intend to bring up in another issue. I haven't seen it non-laggy, yet.)

**Comment by nabiliqbal** [comment]
Thanks -- I am on High Sierra, perhaps that's the issue. (OBS video comes
through for me on Teams as well as on Google Meet, its only an issue with
Zoom.)

**Comment by mikesafar** [comment]
I'm on Catalina 10.15.6.   Everything installed fine, but no virtual camera shows for MS Teams 1.3.00.24758.

**Comment by zeusstl** [comment]
I actually found that the sequence of starting up everything on Zoom and the OBS Virtual Camera actually made a difference in whether the video came through or not.

**Comment by nabiliqbal** [comment]
Thanks, I felt like this made a difference too -- could you tell us exactly which sequence works for you?

**Comment by zeusstl** [comment]
I unfortunately cannot recreate the issue anymore - where before I couldn't get it to work. At some point, during one of my tests, I switched the order I was loading the OBS Virtual Camera and the Zoom video option (turning on OBS and turning on Zoom) and then was able to break and fix it, but unfortunately (or fortunately), now it doesn't happen anymore. Around the same time, I also used Quicktime with OBS which may have also triggered some solution.

I'm sorry I'm not more help, but I can say that it was not working and after fiddling a bunch with the sequence of starting things it started working.

**Comment by mikesafar** [comment]
A few more comments:

1 - I now understand that some apps can see the virtual camera whereas others cannot.  I theorize that this is because the virtual cam depends on an API call that is not being used by all vendors or that they are filtering out virtual cams, maybe for security reasons.   

2 - The delay in the output is ridiculous where it does work (Google Meet).   How much horsepower do you need to make this work?  Granted, I don't have the latest-greatest (2015 Macbook Pro)....just wondering what the expected minimum is.

**Comment by gxalpha** [comment]
@mikesafar 
1) Correct. If you need further information on this, read https://github.com/johnboiles/obs-mac-virtualcam/wiki/Compatibility 
2) That shouldn't be the case. On my 2013 Macbook Pro (pretty base line) it works just fine. Does this happen with all apps (including QuickTime player)? If so, please open a new issue and put in as much info as possible.

@zeusstl @nabiliqbal the order shouldn't matter. Are you using the newest version of this plugin (It should show "1.2.0" on the bottom)?

**Comment by oliverkoetter** [comment]
**Microsoft Teams:**
It does not work with MS Teams for me (with Quicktime it does). On my console I get this volatile error message:

`Library Validation failed: Rejecting '/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam' (Team ID: 62G85M9ZN5, platform: no) for process 'Teams(1479)' (Team ID: UBF8T346G9, platform: no), reason: mapping process and mapped file (non-platform) have different Team IDs`

Maybe that might give an indication on solving this inconsistent problem.

**Comment by mlittle5** [comment]
Hey I was having this problem and I fixed it yesterday! I ran the uninstall script and installed the newest version. I don't know why but this has been working like a charm with zoom and other services.

**Comment by medisoft** [comment]
> Hey I was having this problem and I fixed it yesterday! I ran the uninstall script and installed the newest version. I don't know why but this has been working like a charm with zoom and other services.

Hi! Uninstall of what? OBS or obs-mac-virtualcam? or zoom?

**Comment by mlittle5** [comment]
> > Hey I was having this problem and I fixed it yesterday! I ran the uninstall script and installed the newest version. I don't know why but this has been working like a charm with zoom and other services.
> 
> Hi! Uninstall of what? OBS or obs-mac-virtualcam? or zoom?

Obs virtualcam. Sorry for not being more specific!

**Comment by oliverkoetter** [comment]
I solved it with this fix and it worked perfectly :) Although this may be a MS Teams only solution. Basically it has to do with the signature codesigning of apple (??!?!?!) - simply removing this signature from the involved apps might work - at least with MS Teams it did, might be worth a try for other apps' access to virtual cam.

https://answers.microsoft.com/en-us/msteams/forum/msteams_tfb-msteams_tfmac/microsoft-teams-mac-os-client-is-not-recognizing/d9e863be-d9a4-4d03-a4b8-1b5c7df58828

**Comment by tofutim** [comment]
on zoom the error is

Error loading /Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam:  dlopen(/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam, 262): no suitable image found.  Did find:
	/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam: code signature in (/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam) not valid for use in process using Library Validation: mapped file has no Team ID and is not a platform binary (signed with custom identity or adhoc?)
	/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam: stat() failed with errno=60

also

Cannot find function pointer AppleCMIODPSampleVCamPlugIn for factory <CFUUID 0x6000017be120> 35FDFF29-BFCF-4644-AB77-B759DE932ABE in CFBundle/CFPlugIn 0x7ff08538c540 </Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin> (bundle, not loaded)

**Comment by Nugover** [comment]
I've virtualcam started but it doesn't appears at my Discord. The message error that the console shows is:

FIRST MESSAGE ERROR:
_Error loading /Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam:  dlopen(/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam, 262): no suitable image found.  Did find:
	/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam: code signature in (/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam) not valid for use in process using Library Validation: mapping process and mapped file (non-platform) have different Team IDs_

SECOND MESSAEG ERROR:
_Cannot find function pointer PlugInMain for factory <CFUUID 0x7fccd141b900> 35FDFF29-BFCF-4644-AB77-B759DE932ABE in CFBundle/CFPlugIn 0x7fccd141aa10 </Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin> (bundle, not loaded)_


:(((

**Comment by Rockmoneon** [comment]
Same problem here... It works with QuickTimePlayer, but not with Skype or Teams. I am going crazy.

**Comment by gxalpha** [comment]
@Rockmoneon https://github.com/johnboiles/obs-mac-virtualcam/wiki/Compatibility

**Comment by Rockmoneon** [comment]
> @Rockmoneon https://github.com/johnboiles/obs-mac-virtualcam/wiki/Compatibility

Thank you very much. I've found something like that after hours of searching. I tried with Teams, works great. For Skype I tried the same, but no success. 
But, I created a work around: I downloaded an older version of Skype and installed it. I have to do this every time I start Skype because I can't disable the AutoUpdate. But, it works so far. Thank you once again for your support!
